### PR TITLE
Ensure that run can return non-msgpack serializable results

### PR DESCRIPTION
**Pull request by mrocklin**

### CODE REVIEW

File: distributed/tests/test_client.py [hunk 1]
```diff
@@ -2195,6 +2195,23 @@ def test_run(c, s, a, b):
     assert results == {}
 
 
+@gen_cluster(client=True)
+def test_run_serialize(c, s, a, b):
```
Comment (pitrou) [inline distributed/tests/test_client.py]: Perhaps add a comment and issue number explaining what the test is about?

File: distributed/tests/test_client.py [hunk 2]
```diff
@@ -2195,6 +2195,23 @@ def test_run(c, s, a, b):
...
+    futures = c.map(inc, range(10))
+    yield _wait(futures)
+
+    func = lambda dask_scheduler=None: dask_scheduler.who_has
```
Comment (pitrou) [inline distributed/tests/test_client.py]: Is it required for `dask_scheduler` to be an optional argument?

Comment (mrocklin) [inline distributed/tests/test_client.py]: Apparently no.  Fixed

File: distributed/tests/test_client.py (hunk 1 above)
Comment (mrocklin) [inline distributed/tests/test_client.py]: I've expanded the title to be more informative and changed the code to be, I think, more self explanatory.  There was no related github issue.  The problem was reported in private conversation.

**Comment by pitrou** [comment]
Great, thank you.

**Pull request merged**